src/scenes/CreateEvent/store.js

```diff
@@ -0,0 +1,397 @@
+
+    // Validate hex length
+    const hexString = Web3Utils.toHex(outcome).slice(2); // Remove hex prefix for length validation
+    if (hexString.length > MAX_LEN_RESULT_HEX) {
+      this.error.outcomes[i] = 'create.resultTooLong';
+      return;
+    }
+
+    // Validate cannot name Invalid
+    if (outcome === 'invalid') {
+      this.error.outcomes[i] = 'create.invalidName';
+      return;
+    }
+
+    // Validate no duplicate outcomes
+    const filtered = this.outcomes.filter((item) => (item || '').toLowerCase() === outcome);
+    if (filtered.length > 1) {
+      this.error.outcomes[i] = 'create.duplicateOutcome';
+      return;
+    }
+    this.error.outcomes[i] = '';
+  }
+
+  @action
+  validateResultSetter = async () => {
+    if (await this.isValidAddress()) {
+      this.error.resultSetter = '';
+    } else {
+      this.error.resultSetter = 'create.required';
+    }
+  }
+
+  isValidAddress = async () => {
+    const { data: { result } } = await axios.post(Routes.api.validateAddress, { address: this.resultSetter });
+    return result.isvalid;
+  }
+
+  @action
+  prepareToCreateEvent = async () => {
+    this.validateAll();
+    if (!this.isAllValid) return;
+    const txInfo = {
+      type: TransactionType.APPROVE_CREATE_EVENT,
+      token: Token.BOT,
+      amount: this.escrowAmount,
+      optionIdx: undefined,
+      topicAddress: undefined,
+      oracleAddress: undefined,
+      senderAddress: this.creator,
+    };
+    const { data: { result } } = await axios.post(Routes.api.transactionCost, txInfo);
+    runInAction(() => {
+      this.txFees = result;
+      this.txConfirmDialogOpen = true;
+    });
+    // TODO: do we need this?
+    // const { wallet } = this.app;
+    // if (wallet.needsToBeUnlocked) {
+    //   wallet.unlockDialogOpen = true;
+    // } else {
+    // }
+  }
+
+  validateAll = () => {
+    this.validateTitle();
+    this.validateCreator();
+    this.validatePredictionStartTime();
+    this.validatePredictionEndTime();
+    this.validateResultSettingStartTime();
+    this.validateResultSettingEndTime();
+    for (var i in this.outcomes) { // eslint-disable-line
+      this.validateOutcome(i);
+    }
+    this.validateResultSetter();
+  }
+
+  @action
+  submit = async () => {
+    try {
+      const { data } = await createTopic(
+        this.title,
+        this.outcomes,
+        this.resultSetter,
+        moment(this.prediction.startTime).utc().unix().toString(),
+        moment(this.prediction.endTime).utc().unix().toString(),
+        moment(this.resultSetting.startTime).utc().unix().toString(),
+        moment(this.resultSetting.endTime).utc().unix().toString(),
+        decimalToSatoshi(this.escrowAmount),
+        this.creator, // address
+      );
+      // console.log('CREATE EVENT TX: ', data.createTopic);
+      runInAction(() => {
+        // optimistically add the oracle to the qtum prediction page list
+        // this.app.qtumPrediction.list.push(new Oracle(data.createTopic));
```

Comment:
does this mean the updating does not work currently?

Reply:
yes

Reply:
but now it's fixed